pandas/core/indexes/base.py

```diff
@@ -2316,16 +2316,31 @@ def shift(self, periods=1, freq=None):
 
     def argsort(self, *args, **kwargs):
         """
-        Returns the indices that would sort the index and its
-        underlying data.
+        Return the order of the indices.
+
+        We have an object with values and index. Using argsort method
+        returns a series of indices that would sort the index.
+
+        Parameters
+        ----------
+        *args
+            Passed to `numpy.ndarray.argsort`.
+        **kwargs
+            Passed to `numpy.ndarray.argsort`.
 
         Returns
         -------
-        argsorted : numpy array
+        numpy.ndarray : Argsorted indices of the index
 
         See also
         --------
-        numpy.ndarray.argsort
+        numpy.ndarray.argsort : Returns the indices that would sort this array.
+
+        Examples
+        --------
+        >>> s = pd.Series(data=[4,3,2,1],index=['c','b','a','d'])
```

Comment:
I think you can create the Index directly like `idx = pd.Index(['c', ..])` and then use that

Reply:
ok, I will make two examples, one with series and one with index.